test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -370,18 +374,67 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
 			})
 		})
 	})
+
+	var _ = ginkgo.Context("Requeuing timestamp set to Creation", func() {
+
+		ginkgo.BeforeEach(func() {
+			podsReadyTimeout = 3 * time.Second
+			requeuingTimestamp = config.Creation
+		})
+
+		ginkgo.It("Should prioritize workloads submitted earlier", func() {
+			localQueueName := "eviction-creation-lq"
+
+			// the workloads are created with a 5 cpu resource requirement to ensure only one can fit at a given time
+			wl1 := testing.MakeWorkload("prod1", ns.Name).Queue(localQueueName).Request(corev1.ResourceCPU, "5").Obj()
+			wl2 := testing.MakeWorkload("prod2", ns.Name).Queue(localQueueName).Request(corev1.ResourceCPU, "5").Obj()
+			wl3 := testing.MakeWorkload("prod3", ns.Name).Queue(localQueueName).Request(corev1.ResourceCPU, "5").Obj()
+
+			ginkgo.By("create the workloads", func() {
+				// since metav1.Time has only second resolution, wait one second between
+				// create calls to avoid any potential creation timestamp collision
+				gomega.Expect(k8sClient.Create(ctx, wl1)).Should(gomega.Succeed())
+				time.Sleep(time.Second)
+				gomega.Expect(k8sClient.Create(ctx, wl2)).Should(gomega.Succeed())
+				time.Sleep(time.Second)
```

Comment:
Similarly here, can we capture the essence with just 1 sleep?

Reply:
The expectation here is workload 1 gets admitted (need spacing between 1 & 2), then workload 2 gets admitted (need spacing between 2 & 3 to make this consistent), then workload 1 gets admitted again. I think we need the workloads to be spaced out in time to have trigger this behavior and maintain the conditions being asserted on.

Reply:
sgtm